### PR TITLE
Az module release management

### PR DESCRIPTION
**Pull request by AmrutaKawade**

### CODE REVIEW

File: versions-manifest.json [hunk 1]
```diff
@@ -1,3 +1,11 @@
-[
-    
-]
\ No newline at end of file
+{
+  "version": "0.2.2",
```
Comment (amit-avit) [inline versions-manifest.json]: This file should have an empty array initially as the release was removed from the target branch. You might want to revert the change in this file.

File: Save-ModuleAsZip.ps1 [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[CmdletBinding()]
```
Comment (amit-avit) [inline Save-ModuleAsZip.ps1]: Can we move all the code related files to folder with name src or scripts.

File: .github/workflows/publish_release.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Download and zip Az ps module version ${{ env.VERSION }}
+        if: env.VERSION != ''
+        run: ./Save-ModuleAsZip.ps1 -name Az -version ${{ env.VERSION }} -zipPath ${{ runner.temp }}/az_${{ env.VERSION }}.zip
+        
+      - name: Test if zip is present
+        if: env.VERSION != ''
+        run: |
+          $downloadSucceeded = Test-Path ${{ runner.temp }}/az_${{ env.VERSION }}.zip
+          $downloadSucceeded
```
Comment (amit-avit) [inline .github/workflows/publish_release.yml]: This line #34 is not needed, I had added it to just output the value initially.

File: .github/workflows/publish_release.yml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Finding version to publish
+        run: |
+          $version = ./Find-MinModuleVersionNotInManifest.ps1 -moduleName Az -versionsManifestPath ./versions-manifest.json
```
Comment (amit-avit) [inline .github/workflows/publish_release.yml]: `./scripts/Find-....` as it is moved inside folder.

File: .github/workflows/publish_release.yml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          "VERSION=$version" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf-8 -Append
+
+      - name: Download and zip Az ps module version ${{ env.VERSION }}
+        if: env.VERSION != ''
+        run: ./Save-ModuleAsZip.ps1 -name Az -version ${{ env.VERSION }} -zipPath ${{ runner.temp }}/az_${{ env.VERSION }}.zip
```
Comment (amit-avit) [inline .github/workflows/publish_release.yml]: `./scripts/Save-....` as it is moved inside folder.

File: .github/workflows/publish_release.yml (hunk 4 above)
Comment (amit-avit) [inline .github/workflows/publish_release.yml:23]: We need to remove Find script as now the plan is to cache only the specific versions of the module that we plan to support. So instead take the specific version that we plan to cache as an input.

**Review by chshrikh — APPROVED** [review]
